pkg/odo/cli/component/delete.go

```diff
@@ -74,6 +98,39 @@ func (do *DeleteOptions) Run() (err error) {
 	glog.V(4).Infof("component delete called")
 	glog.V(4).Infof("args: %#v", do)
 
+	if experimental.IsExperimentalModeEnabled() && util.CheckPathExists(do.devfilePath) {
```

Comment:
Another thought I've had: We've been using if blocks in the `Complete/Validate/Run` sections of each CLI command for devfile support, I wonder if we should move them into `devfileComplete/devfileValidate/devfileRun` sections to avoid cluttering the code?

Reply:
Sounds good :+1: I have moved the Run() part for devfiles into a seperate function https://github.com/openshift/odo/pull/2842